postgres/postgres.go

```diff
@@ -545,10 +548,12 @@ func getFilterJobState(tx *sql.Tx, ds Datastore, filterID string) (models.Filter
 	filterJob.FilterID = filterID
 	filterJob.State = state
 
-	if state == submittedState {
-		err := errors.New("Forbidden")
-		log.ErrorC("update to filter job forbidden, job already submitted", err, log.Data{"filter_job_id": filterID})
-		return filterJob, err
+	if !isAuthenticated {
+		if state == submittedState || state == completedState {
+			err := errors.New("Forbidden")
+			log.ErrorC("update to filter job forbidden, job already submitted", err, log.Data{"filter_job_id": filterID})
+			return filterJob, err
```

Comment:
`return nil, err` - no point returning a filterjob that is gonna get ignored because the error returns